### PR TITLE
Feature/fake sensor data

### LINKED ISSUE #122: Fake publisher
Create a fake data publisher. Random temperature data.

### PR DESCRIPTION
**Pull request by TimBuckers**
Fake temperature publisher that will be used to simulate real sensors when running the system only virtual.
- for each sensor a unique topic
- configurable min and max temperature

closes https://github.com/project-march/march-iv/issues/122

Run the node with: `roslaunch march_fake_sensor_data march_fake_sensor_data.launch`
Change parameters real-time: `rosrun rqt_reconfigure rqt_reconfigure`
Visualize the temperature data: `rqt_plot`

### CODE REVIEW

File: march_fake_sensor_data/src/FakeTemperatureData.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+#include <march_fake_sensor_data/TemperaturesConfig.h>
+
+int min_temperature;
+int max_temperature;
+std::vector<std::string> sensor_names;
+std::vector<ros::Publisher> temperature_publishers;
+
+// Store the 7 most recent generated temperatures so we can take the weighted average of them when we publish.
+std::vector<double> latest_temperatures = { 0, 0, 0, 0, 0, 0, 0 };
+
```
Comment (martijnvandermarel) [inline march_fake_sensor_data/src/FakeTemperatureData.cpp:20]: Why doubles?

File: march_fake_sensor_data/src/FakeTemperatureData.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+// The weights of the autoregression, the most recent temperature has the highest weight.
+std::vector<double> ar_values = { 0.1, 0.1, 0.1, 0.15, 0.15, 0.2, 0.2 };
+
+// Calculate the weighted average of the latest_temperatures
+double calculateArTemperature()
+{
```
Comment (martijnvandermarel) [inline march_fake_sensor_data/src/FakeTemperatureData.cpp:26]: Why not give the list as argument for the function?

File: march_fake_sensor_data/src/FakeTemperatureData.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+/**
+ * This callback is called when parameters from the config file are changed during run-time.
+ * This method updates the local values which depend on these parameters to make ensure the values are not out-of-date.
+ * @param config the config file with all the parameters
+ * @param level A bitmask
+ */
+void callback(march_fake_sensor_data::TemperaturesConfig& config, uint32_t level)
+{
```
Comment (martijnvandermarel) [inline march_fake_sensor_data/src/FakeTemperatureData.cpp:42]: 'callback' too vague. More specific name?

**Review by martijnvandermarel — APPROVED** [review]

**Review by Ishadijcks — APPROVED** [review]
Looks good, auto-regression is awesome.

**Review by sopje — APPROVED** [review]

**Pull request merged**